tests/integration/test_ask_tell_optimization.py

```diff
@@ -164,8 +199,13 @@ def build_model(data: Dataset) -> GaussianProcessRegression:
     best_y = dataset.observations[arg_min_idx]
     best_x = dataset.query_points[arg_min_idx]
 
-    relative_minimizer_err = tf.abs((best_x - BRANIN_MINIMIZERS) / BRANIN_MINIMIZERS)
-    # these accuracies are the current best for the given number of optimization steps, which makes
-    # this is a regression test
-    assert tf.reduce_any(tf.reduce_all(relative_minimizer_err < 0.05, axis=-1), axis=0)
-    npt.assert_allclose(best_y, SCALED_BRANIN_MINIMUM, rtol=0.005)
+    if branin:
+        relative_minimizer_err = tf.abs((best_x - BRANIN_MINIMIZERS) / BRANIN_MINIMIZERS)
+        # these accuracies are the current best for the given number of optimization steps,
+        # which makes this is a regression test
+        assert tf.reduce_any(tf.reduce_all(relative_minimizer_err < 0.05, axis=-1), axis=0)
+        npt.assert_allclose(best_y, SCALED_BRANIN_MINIMUM, rtol=0.005)
+    else:
+        # currently assume that every rule should be able to solve this in 5 steps
+        npt.assert_allclose(best_x, tf.constant([1.0, 1.0], tf.float64), rtol=0.05)
+        npt.assert_allclose(best_y, tf.constant([-4.0], tf.float64), rtol=0.05)
```

Comment:
Again, would be nice to put these to constants and move along the function definition. In fact, maybe it's worth having this funciton just as any other objective in `single_objectives`?